autorag/nodes/passagereranker/colbert.py

```diff
@@ -35,26 +34,29 @@ def colbert_reranker(queries: List[str], contents_list: List[List[str]],
     model = AutoModel.from_pretrained(model_name).to(device)
     tokenizer = AutoTokenizer.from_pretrained(model_name)
 
-    # Run async cohere_rerank_pure function
-    tasks = [get_colbert_score(query, document, model, tokenizer) for query, document, ids in
-             zip(queries, contents_list, ids_list)]
-    loop = asyncio.get_event_loop()
-    score_results = loop.run_until_complete(process_batch(tasks, batch_size=batch))
+    rerank_scores = colbert_run_model(queries, contents_list, model, tokenizer, batch_size=batch)
+
+    sorted_contents, sorted_ids, sorted_scores = sort_and_select_top_k(contents_list, ids_list, rerank_scores, top_k)
 
     del model
+    if torch.cuda.is_available():
+        torch.cuda.empty_cache()
+
+    return sorted_contents, sorted_ids, sorted_scores
+
+
+def colbert_run_model(queries, contents_list, model, tokenizer, batch_size: int):
+    batch_queries_list = make_batch(queries, batch_size)
+    batch_contents_list = make_batch(contents_list, batch_size)
 
-    def rerank_results(contents, ids, scores, top_k):
-        reranked_content, reranked_id, reranked_score = zip(
-            *sorted(zip(contents, ids, scores), key=lambda x: x[2], reverse=True))
-        return list(reranked_content)[:top_k], list(reranked_id)[:top_k], list(reranked_score)[:top_k]
+    results = list(map(lambda pair: get_colbert_score(*pair, model, tokenizer),
+                       zip(sum(batch_queries_list, []), sum(batch_contents_list, []))))
```

Comment:
input of get_colbert_score related to query is 'query: str', and It seems not to work in parallel in my idea. it works in parallel?